app/main/db/queries.py

```diff
@@ -288,19 +302,108 @@ def _build_consignment_filter_query(consignment_id: uuid.UUID):
                 ),
                 else_=None,
             ),
-        ),
+        ).label("closure_period"),
     )
 
     filters = [
         File.ConsignmentId == consignment_id,
         func.lower(File.FileType) == "file",
     ]
 
-    query = (
+    sub_query = (
         select.join(FileMetadata, File.FileId == FileMetadata.FileId)
         .filter(*filters)
         .group_by(File.FileId)
         .order_by(File.FileName)
+    ).subquery()
+
+    query = db.session.query(
+        sub_query.c.file_id,
+        sub_query.c.file_name,
+        func.to_char(
+            sub_query.c.date_last_modified,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("date_last_modified"),
+        sub_query.c.closure_type,
+        func.to_char(
+            sub_query.c.closure_start_date,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("closure_start_date"),
+        sub_query.c.closure_period,
     )
 
+    dt_range = get_date_range(date_range)
```

Comment:
After some refactoring and changing from passing in date_range dict to date_from and date_to we can simplify this to

```
    if date_from and date_to:
        query = and_(
                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
                >= date_from,
                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
                <= date_to,
            )
    elif date_from:
        query = func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD") >= date_from
    elif date_to:
        date_filter = func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD") <= date_to

    query = query.filter(date_filter)
```

Reply:
modified logic in queries.py , created a new function _build_date_range_filter , it can be used for date range parameter